### PR TITLE
EIP-1283 improvements

### PR DESCRIPTION
**Pull request by ayrat555**
Changes:
- Clean cache code
- Add eip1283 tests from  https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1283.md#test-cases
    - fix found bug
- Clean refunds, gas cost code

### CODE REVIEW

File: apps/evm/lib/evm/refunds/sstore.ex [hunk 1]
```diff
@@ -35,23 +39,35 @@ defmodule EVM.Refunds.Sstore do
...
+      initial_value == current_value && new_value == 0 ->
         15_000
 
-      initial_value != current_value && initial_value != 0 && current_value == 0 ->
-        -15_000
+      true ->
+        first_refund =
+          cond do
+            initial_value != 0 && current_value == 0 ->
+              -15_000
```
Comment (masonforest) [inline apps/evm/lib/evm/refunds/sstore.ex]: can we pull these out into module variables like we did with the other gas values?

File: apps/blockchain/test/blockchain/contract/message_call_test.exs [hunk 2]
```diff
@@ -160,4 +160,322 @@ defmodule Blockchain.Contract.MessageCallTest do
       assert output == :failed
     end
   end
+
+  # https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1283.md#test-cases
```
Comment (germsvel) [inline apps/blockchain/test/blockchain/contract/message_call_test.exs]: 🎉

File: mix.exs [hunk 3]
```diff
@@ -38,7 +38,7 @@ defmodule Mana.MixProject do
   defp deps do
     [
       {:ex_doc, "~> 0.19", only: :dev, runtime: false},
-      {:dialyxir, "~> 1.0.0-rc.3", only: [:dev, :test], runtime: false},
```
Comment (germsvel) [inline mix.exs]: Any reason why this change is in here? I think Dialyxir's new warnings are far better.

Comment (MattMSumner) [reply mix.exs]: If you downgraded to get the output to put into the ignore file, on the newer version you can run `mix dialyzer --format dialyzer` to get the old format.

File: apps/evm/lib/evm/refunds/sstore.ex (hunk 1 above)
Comment (ayrat555) [reply apps/evm/lib/evm/refunds/sstore.ex]: done

File: mix.exs (hunk 3 above)
Comment (ayrat555) [reply mix.exs]: yep, exactly that was a problem for me. But I returned the new version
